src/Fixer/Casing/NativeTypeDeclarationCasingFixer.php

```diff
@@ -114,52 +49,40 @@ final class NativeTypeDeclarationCasingFixer extends AbstractFixer
      *
      * @var array<string, true>
      */
-    private array $functionTypeHints;
-
-    private FunctionsAnalyzer $functionsAnalyzer;
-
-    /**
-     * @var list<array{int}|string>
-     */
-    private array $beforePropertyTypeTokens;
+    private array $types;
 
     public function __construct()
     {
         parent::__construct();
 
-        $this->beforePropertyTypeTokens = ['{', ';', [T_PRIVATE], [T_PROTECTED], [T_PUBLIC], [T_VAR]];
-
-        $this->functionTypeHints = [
+        $this->types = [
             'array' => true,
             'bool' => true,
             'callable' => true,
             'float' => true,
             'int' => true,
             'iterable' => true,
+            'mixed' => true,
             'object' => true,
+            'parent' => true,
             'self' => true,
+            'static' => true,
             'string' => true,
             'void' => true,
         ];
 
         if (\PHP_VERSION_ID >= 8_00_00) {
-            $this->functionTypeHints['false'] = true;
-            $this->functionTypeHints['mixed'] = true;
```

Comment:
why allowing mixed/static for 7.4 ?

Reply:
- `mixed` - because I have trusted the tests, I've reverted the change - take a look at the latest commit
- `static` - because we [support](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.68.1/tests/Fixer/Casing/NativeTypeDeclarationCasingFixerTest.php#L252-L255) this type - when "type" in this case means type of property (dynamic or static). In PHP 7.4 `static` is not allowed as a parameter/return type, but then it is a syntax error, so the fixer won't fix it anyway.

Reply:
i would be careful to trust the tests :D thanks